timer.R

```diff
@@ -0,0 +1,36 @@
+library(tictoc)
+
+printTicTocLog <-
+  function() {
+    tic.log() %>%
+      unlist %>%
+      tibble(logvals = .) %>%
+      separate(logvals,
+               sep = ":",
+               into = c("Function type", "log")) %>%
+      mutate(log = str_trim(log)) %>%
+      separate(log,
+               sep = " ",
+               into = c("Seconds"),
+               extra = "drop")
+  }
+
+tic.clearlog()
+
+tic("Run Script1")
+source("scripts/script1.R")
+toc(log = TRUE)
+
+tic("Run Script2")
+source("scripts/script2.R")
+toc(log = TRUE)
+
+tic("Run Script3")
+source("scripts/script3.R")
+toc(log = TRUE)
+
+printTicTocLog() %>%
+  knitr::kable()
+
+# For me, Script1 took 66.04 seconds, Script2 took 56.72 seconds 
+# and Script 3 took 49.05. This means that Script3 is the fastest of the three. 
```

Comment:
Good job! The reason that the last one is quicker than the second one is because the second one parallelizes the rows in df. Because there are only 5 rows, having more than 5 cores will not make it quicker. The last one parallizes M, and can therefore utlize more cores in the computation